### ISSUE TITLE
org.jboss.as.arquillian.container.managed.ManagedDeployableContainer should not be final

### ISSUE THREAD

**Issue by chrisruffalo** [wildfly/wildfly-arquillian#156]
I was trying to customize the start of my Wildfly (EAP7) container and handle some before-start tasks and I could not extend the ManagedDeployableContainer class which complicates this task.

Please make the class non-final unless there is a compelling reason to prevent extension.

**Comment by jamezp** [comment]
What kind of tasks were you trying to perform? There is likely a workaround.

**Comment by chrisruffalo** [comment]
I want to run JBoss CLI files against the Chameleon instance. I have files we used to build our Docker images and want to re-use the same files. Another option was to use the SetupTask and then use the management client. The result has been better to invoke the CLI client bundled with EAP/WildFly directly against our configuration artifacts.

All of this reduces the chance of configuration drift and ensures that our docker images have matching configurations. (Highly desirable to me.)

**Comment by jamezp** [comment]
It's been a year so I assume this as been resolved. I apologize for not getting back to you earlier. I think I stopped seeing notifications here for some reason as there are several issues I see that need attention. I've filed https://issues.redhat.com/browse/WFARQ-71 which may somewhat resolve this.

The `ServerSetup` is really designed to configure the server. The main reason is the tearDown() may be needed for subsequent tests.
